src/allencell_ml_segmenter/thresholding/thresholding_service.py

```diff
@@ -48,83 +44,97 @@ def __init__(
         self._task_executor: ITaskExecutor = task_executor
 
         self._thresholding_model.subscribe(
-            Event.ACTION_THRESHOLDING_VALUE_CHANGED,
+            Event.ACTION_EXECUTE_THRESHOLDING,
             self,
             self._on_threshold_changed,
         )
 
         self._thresholding_model.subscribe(
-            Event.ACTION_THRESHOLDING_AUTOTHRESHOLDING_SELECTED,
+            Event.ACTION_SAVE_THRESHOLDING_IMAGES,
             self,
-            self._on_threshold_changed,
+            self._save_thresholded_images,
         )
 
         self._thresholding_model.subscribe(
-            Event.ACTION_SAVE_THRESHOLDING_IMAGES,
+            Event.ACTION_THRESHOLDING_DISABLED,
             self,
-            self._save_thresholded_images,
+            self._remove_all_prob_maps,
         )
 
     def _handle_thresholding_error(self, error: Exception) -> None:
         show_info("Thresholding failed: " + str(error))
 
     def _on_threshold_changed(self, _: Event) -> None:
-        layers_containing_prob_map: list[Layer] = (
-            self._viewer.get_all_layers_containing_prob_map()
-        )
+        # Check to see if user has selected a thresholding method
+        if (
```

Comment:
A lot of logic would be simplified by 

- not having a 'none' option
- having a default selection (auto-thresholding seems like a good default)